recipes/libigl/all/conan_deps.cmake

```diff
@@ -0,0 +1 @@
+find_package(Eigen3 REQUIRED)
```

Comment:
Could you explain and document why this is needed? First time I see this approach in a recipe

Reply:
It's pretty much a direct replacement for https://github.com/conan-io/conan-center-index/pull/18849/files#diff-81c0c5b750ed5544831cb54fb1e24013a3566ab140ffcbaa87c12e50f32e2931

It's not that relevant in this PR, but the `tc.variables["CMAKE_PROJECT_libigl_INCLUDE"] = "conan_deps.cmake"` pattern is something that I have used in several PRs to have tighter control over `find_package()` without needing to do a ton of patching in the CMake files. An example: https://github.com/conan-io/conan-center-index/pull/18900/files#diff-8175d97eafddd9bb81bd8992d385ce6790d86eef27d291becbbe687fd7bc660d